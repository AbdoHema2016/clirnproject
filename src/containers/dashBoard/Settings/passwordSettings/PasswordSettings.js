import React from 'react';
import {
  View,
  KeyboardAvoidingView,
  ScrollView,
  Platform,
  Switch,
  Text,
} from 'react-native';
import Style from './Style';
import {connect} from 'react-redux';
import CLabel from '../../../../components/cLabel';
import CButton from '../../../../components/cButton';
import navigationService from '../../../../Navigation/NavigationService';
import CNavigationBackButton from '../../../../components/cNavigationBackButton';
import {HelperFunctions, testIds} from '../../../../utilities';
import {translate} from '../../../../Localization';

import {updatePasswordAction} from './redux/actions';
import HelperFunction from './helperFunctions';
import {
  sendPublicKeyAction,
  updatePublicKeyAction,
} from '../../../Biomitric/redux/actions';
import {checkIfBiometricIdIsSupported} from '../../../Biomitric/Biometrics';

const {showErrorMessage} = HelperFunctions;

class PasswordSettings extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      oldPassword: {value: '', state: '', hidden: true},
      newPassword: {value: '', state: '', hidden: true},
      confirmNewPassword: {value: '', state: '', hidden: true},
      checkError: false,
      biometricEnabled: false,
      biometricAvailable: false,
    };
    HelperFunction.setProps = this.props;
    HelperFunction.setInstance = this;
    navigationService.navigation = this.props.navigation;
  }
  componentDidMount() {
    this.navigationOptions();
    HelperFunction.checkBiometricEnabled();
    this.checkifBiometricAvailable();
  }
  checkifBiometricAvailable = async () => {
    try {
      let biometricAvailable = await checkIfBiometricIdIsSupported();
      if (biometricAvailable) {
        this.setState({biometricAvailable: true});
      }
    } catch (error) {
      showErrorMessage(translate('STRINGS.SOMETHING_WENT_WRONG'));
    }
  };
  navigationOptions = () => {
    this.props.navigation.setOptions({
      title: '',
      headerTransparent: true,
      headerLeft: () => (
        <CNavigationBackButton
          grey={true}
          title={translate('BACK.BACK_BUTTON_TITLE')}
          titleStyle={Style.backButtonStyle}
          backButtonAction={this.goBack}
        />
      ),
    });
  };
  goBack = () => {
    this.props.navigation.pop();
  };
  goToForgotPasswordScreen = () => {
    this.props.navigation.navigate('ForgotPassword', {isLoggedInUser: true});
  };
  render() {
    let {biometricAvailable} = this.state;
    return (
      <View style={Style.container}>
        <KeyboardAvoidingView
          style={Style.keyboardAvoidingView}
          behavior={Platform.OS === 'ios' ? 'padding' : null}
          keyboardVerticalOffset={Style.keyboardVerticalOffset.top}>
          <ScrollView
            bounces={false}
            ref={this.scrollView}
            alwaysBounceVertical={false}
            showsVerticalScrollIndicator={false}>
            <View style={Style.innerContainer}>
              <CLabel
                style={Style.appTitle}
                text={translate('PASSWORD_SETTINGS.PASSWORD_TITLE')}
              />
              <View style={Style.inputContainer}>
                {HelperFunction.createPasswordFeilds()}
              </View>
            </View>
            <View style={Style.loginViaFaceIDView}>
              {biometricAvailable && (
                <>
                  <CLabel
                    style={Style.loginViaFaceIDLabel}
                    text={translate('PASSWORD_SETTINGS.LOGIN_USING_BIOMETRIC')}
                  />
                  <Switch
                    style={Style.switch}
                    onValueChange={HelperFunction.changeBiometric}
                    value={this.state.biometricEnabled}
                  />
                </>
              )}
            </View>
          </ScrollView>
        </KeyboardAvoidingView>
        <View style={Style.passwordbuttonsContainer}>
          {!this.state.oldPassword.value ? (
            <Text
              style={Style.forgotPasswordLabel}
              onPress={this.goToForgotPasswordScreen}>
              {translate('LOGIN_SCREEN_STRINGS.PASSWORD_FORGOTTEN')}
            </Text>
          ) : (
            <CButton
              accessibilityLabel={testIds.updatePasswordBtn}
              testID={testIds.updatePasswordBtn}
              text={translate('CONTACT_DETAILS_SCREEN_STRINGS.UPDATE')}
              textStyle={Style.whiteText}
              onPress={HelperFunction.updatePassword}
              buttonContainerStyle={Style.blackButton}
              loading={this.props.loading}
              backgroundColor={Style.blackButton.backgroundColor}
            />
          )}
        </View>
      </View>
    );
  }
}
const mapStateToProps = (state) => {
  const {
    changePassword: {loading},
    signIn: {userID, token},
    userProfile: {
      userInfo: {email},
    },
  } = state;
  return {
    userID,
    access_token: token,
    loading,
    email,
  };
};
const mapDispatchToProps = {
  updatePassword: updatePasswordAction,
  sendPublicKey: sendPublicKeyAction,
  updatePublicKey: updatePublicKeyAction,
};
export default connect(mapStateToProps, mapDispatchToProps)(PasswordSettings);
