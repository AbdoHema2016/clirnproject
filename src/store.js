import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
