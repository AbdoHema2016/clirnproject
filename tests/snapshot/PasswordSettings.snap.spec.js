import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import PasswordSettings from '../../src/containers/dashBoard/Settings/passwordSettings/PasswordSettings';

jest.mock('@react-navigation/native', () => {});
jest.mock('react-native-gesture-handler', () => {});
jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('Password settings screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('Snapshot Password settings', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PasswordSettings {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
