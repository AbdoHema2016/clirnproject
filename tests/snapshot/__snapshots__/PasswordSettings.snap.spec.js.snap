// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password settings screen Snapshot Password settings 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      alwaysBounceVertical={false}
      bounces={false}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Object {
              "justifyContent": "space-around",
              "width": 724,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2B342C",
                  "fontFamily": "SofiaPro-Medium",
                  "fontSize": 24,
                  "fontWeight": "600",
                  "marginTop": 200.1,
                },
              ]
            }
          >
            [missing "en.PASSWORD_SETTINGS.PASSWORD_TITLE" translation]
          </Text>
          <View
            style={
              Object {
                "height": "auto",
                "marginTop": 20,
                "width": 724,
              }
            }
          >
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2B342C",
                      "fontFamily": "SofiaProRegular",
                      "fontSize": 12,
                      "height": 16,
                      "marginTop": 16,
                    },
                  ]
                }
              >
                [missing "en.PASSWORD_SETTINGS.CURRENT_PASSWORD" translation]
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "height": "auto",
                      "width": 724,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "height": 48,
                        "justifyContent": "center",
                        "marginTop": 4,
                        "width": "100%",
                      },
                      Object {
                        "backgroundColor": "#F7F7F7",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "marginTop": 4,
                      },
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="oldPassword"
                    allowFontScaling={true}
                    blurOnSubmit={false}
                    editable={true}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="[missing \\"en.PASSWORD_SETTINGS.ENTER_THE_PASSWORD\\" translation]"
                    placeholderTextColor="#8D8C91"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    secureTextEntry={true}
                    style={
                      Array [
                        Object {
                          "color": "#2B342C",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "lineHeight": 20,
                          "paddingLeft": 19,
                          "paddingRight": 19,
                          "width": "100%",
                        },
                        Object {
                          "marginLeft": 0,
                          "width": "90%",
                        },
                      ]
                    }
                    testID="oldPassword"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <View
                    accessibilityLabel="Eye"
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="Eye"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "height": 25,
                            "marginRight": 18,
                            "width": 25,
                          },
                        ]
                      }
                    >
                      <FastImageView
                        resizeMode="cover"
                        source={
                          Object {
                            "cache": "immutable",
                            "uri": undefined,
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2B342C",
                      "fontFamily": "SofiaProRegular",
                      "fontSize": 12,
                      "height": 16,
                      "marginTop": 16,
                    },
                  ]
                }
              >
                [missing "en.PASSWORD_SETTINGS.NEW_PASSWORD" translation]
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "height": "auto",
                      "width": 724,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "height": 48,
                        "justifyContent": "center",
                        "marginTop": 4,
                        "width": "100%",
                      },
                      Object {
                        "backgroundColor": "#F7F7F7",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "marginTop": 4,
                      },
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="newPassword"
                    allowFontScaling={true}
                    blurOnSubmit={false}
                    editable={true}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="[missing \\"en.PASSWORD_SETTINGS.ENTER_NEW_PASSWORD\\" translation]"
                    placeholderTextColor="#8D8C91"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    secureTextEntry={true}
                    style={
                      Array [
                        Object {
                          "color": "#2B342C",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "lineHeight": 20,
                          "paddingLeft": 19,
                          "paddingRight": 19,
                          "width": "100%",
                        },
                        Object {
                          "marginLeft": 0,
                          "width": "90%",
                        },
                      ]
                    }
                    testID="newPassword"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <View
                    accessibilityLabel="Eye"
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="Eye"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "height": 25,
                            "marginRight": 18,
                            "width": 25,
                          },
                        ]
                      }
                    >
                      <FastImageView
                        resizeMode="cover"
                        source={
                          Object {
                            "cache": "immutable",
                            "uri": undefined,
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2B342C",
                      "fontFamily": "SofiaProRegular",
                      "fontSize": 12,
                      "height": 16,
                      "marginTop": 16,
                    },
                  ]
                }
              >
                [missing "en.PASSWORD_SETTINGS.CONFIRM_NEW_PASSWORD" translation]
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "height": "auto",
                      "width": 724,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "height": 48,
                        "justifyContent": "center",
                        "marginTop": 4,
                        "width": "100%",
                      },
                      Object {
                        "backgroundColor": "#F7F7F7",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "marginTop": 4,
                      },
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="confirmPassword"
                    allowFontScaling={true}
                    blurOnSubmit={false}
                    editable={true}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="[missing \\"en.PASSWORD_SETTINGS.ENTER_NEW_PASSWORD\\" translation]"
                    placeholderTextColor="#8D8C91"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    secureTextEntry={true}
                    style={
                      Array [
                        Object {
                          "color": "#2B342C",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "lineHeight": 20,
                          "paddingLeft": 19,
                          "paddingRight": 19,
                          "width": "100%",
                        },
                        Object {
                          "marginLeft": 0,
                          "width": "90%",
                        },
                      ]
                    }
                    testID="confirmPassword"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <View
                    accessibilityLabel="Eye"
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="Eye"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "height": 25,
                            "marginRight": 18,
                            "width": 25,
                          },
                        ]
                      }
                    >
                      <FastImageView
                        resizeMode="cover"
                        source={
                          Object {
                            "cache": "immutable",
                            "uri": undefined,
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 28,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "bottom": 157.41199999999998,
        "position": "absolute",
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "bottom": 44,
          "color": "#02BF8E",
          "fontFamily": "SofiaProRegular",
          "fontSize": 18,
          "height": 20,
        }
      }
    >
      [missing "en.LOGIN_SCREEN_STRINGS.PASSWORD_FORGOTTEN" translation]
    </Text>
  </View>
</View>
`;
